src/System.Data.SQLite/SQLiteDataReader.cs

```diff
@@ -50,9 +54,43 @@ private Task<bool> NextResultAsyncCore(CancellationToken cancellationToken)
 					int index;
 					if (parameterName is not null)
 					{
+#if NET5_0
+						static unsafe int GetParameterIndex(SqliteStatementHandle currentStatement, string parameterName)
+						{
+							var utf8Length = Encoding.UTF8.GetByteCount(parameterName) + 2;
+
+							const int c_stackAllocationThreshold = 1024;
```

Comment:
Almost all parameter names should be (much) less than 1024 characters, so this check is probably overkill. Could just always `stackalloc` and blow up if the parameter name length smashed the stack.